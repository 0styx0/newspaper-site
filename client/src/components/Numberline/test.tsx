import * as React from 'react';
import { MemoryRouter } from 'react-router';
import * as renderer from 'react-test-renderer';
import NumberlineContainer from './container';

describe('<NumberlineContainer>', () => {

    describe('snapshots', () => {

        function snap(current: number, max: number) {

            const tree = renderer.create(
                <MemoryRouter>
                    <NumberlineContainer current={current} max={max} />
                </MemoryRouter>
            ).toJSON();
            expect(tree).toMatchSnapshot();
        }

        it('renders correctly with 5 numbers', () => snap(3, 5));

        it('with more than 5 makes an ellipses', () => snap(5, 20));

        it('with current in the middle of line', () => snap(20, 40));
    });
});
