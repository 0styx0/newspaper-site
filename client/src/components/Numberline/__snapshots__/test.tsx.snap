// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NumberlineContainer> snapshots renders correctly with 5 numbers 1`] = `
<span
  id="issueRange"
>
  <a
    href="/issue/1"
    onClick={[Function]}
  >
    1
  </a>
  <a
    href="/issue/2"
    onClick={[Function]}
  >
    2
  </a>
  <a
    href="/issue/3"
    onClick={[Function]}
  >
    3
  </a>
  <a
    href="/issue/4"
    onClick={[Function]}
  >
    4
  </a>
  <a
    href="/issue/5"
    onClick={[Function]}
  >
    5
  </a>
</span>
`;

exports[`<NumberlineContainer> snapshots with current in the middle of line 1`] = `
<span
  id="issueRange"
>
  <a
    href="/issue/1"
    onClick={[Function]}
  >
    1
  </a>
  <a
    href="/issue/2"
    onClick={[Function]}
  >
    2
  </a>
  <a
    href="/issue/3"
    onClick={[Function]}
  >
    3
  </a>
  <a
    href="/issue/4"
    onClick={[Function]}
  >
    4
  </a>
  <a
    href="/issue/5"
    onClick={[Function]}
  >
    5
  </a>
  ...
  <a
    href="/issue/18"
    onClick={[Function]}
  >
    18
  </a>
  <a
    href="/issue/19"
    onClick={[Function]}
  >
    19
  </a>
  <a
    href="/issue/20"
    onClick={[Function]}
  >
    20
  </a>
  <a
    href="/issue/21"
    onClick={[Function]}
  >
    21
  </a>
  <a
    href="/issue/22"
    onClick={[Function]}
  >
    22
  </a>
  <a
    href="/issue/23"
    onClick={[Function]}
  >
    23
  </a>
  ...
  <a
    href="/issue/38"
    onClick={[Function]}
  >
    38
  </a>
  <a
    href="/issue/39"
    onClick={[Function]}
  >
    39
  </a>
  <a
    href="/issue/40"
    onClick={[Function]}
  >
    40
  </a>
</span>
`;

exports[`<NumberlineContainer> snapshots with more than 5 makes an ellipses 1`] = `
<span
  id="issueRange"
>
  <a
    href="/issue/1"
    onClick={[Function]}
  >
    1
  </a>
  <a
    href="/issue/2"
    onClick={[Function]}
  >
    2
  </a>
  <a
    href="/issue/3"
    onClick={[Function]}
  >
    3
  </a>
  <a
    href="/issue/4"
    onClick={[Function]}
  >
    4
  </a>
  <a
    href="/issue/5"
    onClick={[Function]}
  >
    5
  </a>
  <a
    href="/issue/6"
    onClick={[Function]}
  >
    6
  </a>
  <a
    href="/issue/7"
    onClick={[Function]}
  >
    7
  </a>
  <a
    href="/issue/8"
    onClick={[Function]}
  >
    8
  </a>
  ...
  <a
    href="/issue/18"
    onClick={[Function]}
  >
    18
  </a>
  <a
    href="/issue/19"
    onClick={[Function]}
  >
    19
  </a>
  <a
    href="/issue/20"
    onClick={[Function]}
  >
    20
  </a>
</span>
`;
